### PR TITLE
Fix tightly-coupled dependency on Stylelint's internal module `lib/utils/getOsEol`

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Ref https://github.com/stylelint/stylelint/pull/7099

> Is there anything in the PR that needs further explanation?

This change aims to make changes easier by removing the dependency on an internal module.

Summary:
- Change the `os.EOL` property by the before/all hooks instead of mocking.
- Add a file for `setupFilesAfterEnv`.
- Delay requiring `stylelint` in `getTestRule()`.

See also:
- https://github.com/stylelint/stylelint/blob/0dbed54626e87786c2ef2e4ccd4c27d04c8a6918/lib/utils/getOsEol.js
- https://jestjs.io/docs/configuration/#setupfilesafterenv-array

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## Head
+
+- Removed: dependency on Stylelint's internal module `lib/utils/getOsEol`.
```
Comment (ybiquitous) [inline CHANGELOG.md:5]: [ask] Should we release this change as a *breaking change*?

File: jest-setup-after-env.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+beforeAll(() => {
+	// NOTE: `jest.replaceProperty()` is unavailable for a read-only property.
+	Object.defineProperty(os, 'EOL', { ...eolDescriptor, value: '\n' });
+});
+
+afterAll(() => {
+	Object.defineProperty(os, 'EOL', eolDescriptor);
+});
```
Comment (ybiquitous) [inline jest-setup-after-env.js:18]: [ask] We may be able to move these hooks for `os.EOL` into `testRule()` like this. Any thoughts?

```js
return function testRule(schema) {
	/** @type {import('stylelint').lint} */
	let lint;
	const eolDescriptor = Object.getOwnPropertyDescriptor(os, 'EOL');

	beforeAll(() => {
		// NOTE: `jest.replaceProperty()` is unavailable for a read-only property.
		Object.defineProperty(os, 'EOL', { ...eolDescriptor, value: '\n' });

		// eslint-disable-next-line n/no-unpublished-require -- Avoid auto-install of `stylelint` peer dependency.
		lint = require('stylelint').lint;
	});

	afterAll(() => {
		Object.defineProperty(os, 'EOL', eolDescriptor);
	});
```

Comment (jeddy3) [reply jest-setup-after-env.js:18]: I think it looks good where it is.

File: CHANGELOG.md (hunk 1 above)
Comment (jeddy3) [reply CHANGELOG.md:5]: Is it breaking for the user? It seems closer to a refactor. How about patch?

> - Fixed: tightly-coupled dependency on Stylelint's internal module `lib/utils/getOsEol`.

Comment (ybiquitous) [reply CHANGELOG.md:5]:
```suggestion
- Fixed: tightly-coupled dependency on Stylelint's internal module `lib/utils/getOsEol`.
```

File: jest-setup-after-env.js (hunk 2 above)
Comment (ybiquitous) [reply jest-setup-after-env.js]: Thanks. Fixed ebe4650.

Comment (ybiquitous) [reply jest-setup-after-env.js]: Oops. ebe4650 doesn't work expectedly 😢 

https://github.com/stylelint/stylelint/actions/runs/5666393030/job/15353040831?pr=7099#step:7:4555

Comment (ybiquitous) [reply jest-setup-after-env.js:18]: Reverted 3310063

**Review by jeddy3 — APPROVED** [review]
LGTM, thank you!

**Comment by ybiquitous** [comment]
Thanks for the review. I'll release 6.1.1 soon.

**Comment by ybiquitous** [comment]
https://github.com/stylelint/jest-preset-stylelint/releases/tag/6.1.1 released 🎉

**Pull request merged**